<?xml version="1.1" encoding="UTF-8" standalone="yes"?>
<launch>
	<!-- sim time -->
	<param name="/use_sim_time" value="true" />
	<arg name="nodelet_manager" default="nodelet_manager" />

	<!-- static transform publisher in sim time -->
	<node pkg="tf" type="static_transform_publisher" name="lidar2base_publisher" args="0 0 0.45 0 0 0 base_link velodyne 10" />

	<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/>

	<!-- <node pkg="magni_octonav" type="cloud_projector_node" name="cloud_projector_node" output="screen" > -->
	<node pkg="nodelet" type="nodelet" name="cloud_projector_nodelet" args="load magni_octonav/CloudProjectorNodelet $(arg nodelet_manager)" >
		<param name="base_frame" value="base_link" />
		<param name="sensor_frame" value="velodyne" />

		<!-- VLP16 parameters -->
		<param name="angle_resolution" value="0.00347435014745802414" />
		<param name="beam_resolution" value="0.03490658503988659154" />
		<param name="channel_size" value="16" />
	</node>

	<node pkg="lego_loam" type="imageProjection" name="imageProjection" output="screen"/>
	
</launch>